File: forgerock-auth/src/main/java/org/forgerock/android/auth/Token.java
/*
 * Copyright (c) 2019 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

/**
 * Domain object to hold generic Token
 */
@EqualsAndHashCode
@AllArgsConstructor
@Getter
public class Token implements Serializable {
    private String value;

}

File: forgerock-auth/src/main/java/org/forgerock/android/auth/Token.kt
/*
 * Copyright (c) 2022 - 2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */
package org.forgerock.android.auth

import kotlinx.serialization.Serializable

/**
 * Domain object to hold generic Token
 */
@Serializable
sealed interface Token {
    val value: String
}